lib/Index/Index.cpp

```diff
@@ -189,15 +189,21 @@ class IndexSwiftASTWalker : public SourceEntityWalker {
     return false;
   }
 
-  bool addRelation(IndexSymbol &Info, SymbolRoleSet RelationRoles, ValueDecl *D) {
+  bool addRelation(IndexSymbol &Info, SymbolRoleSet RelationRoles, Decl *D) {
     assert(D);
     StringRef Name, USR;
     SymbolInfo SymInfo = getSymbolInfoForDecl(D);
 
     if (SymInfo.Kind == SymbolKind::Unknown)
       return true;
-    if (getNameAndUSR(D, /*ExtD=*/nullptr, Name, USR))
-      return true;
+    if (auto *ExtD = dyn_cast<ExtensionDecl>(D)) {
+      NominalTypeDecl *NTD = ExtD->getExtendedType()->getAnyNominal();
```

Comment:
No, in such a case `IndexSwiftASTWalker::reportExtension` would bail out early and `addRelation` would not get called.